### PR TITLE
Modify abseil cmake to not treat warnings as errors

### PR DESCRIPTION
**Pull request by linnealovespie**
**Description**: Describe your changes.
Moved abseil-cpp.cmake to its own folder abseil-cpp and add the /WX- compile option so that compile warnings from abseil-cpp aren't treated as errors. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
WinML treats warnings as errors and [f87a06c](https://github.com/microsoft/onnxruntime/commit/f87a06cd9617ab67d06e57e4c867e799f0d1c255) raises warnings for abseil. This change keeps warnings for abseil but stops winml build from breaking.  
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Review by tiagoshibata — APPROVED** [review]

**Review by tiagoshibata — APPROVED** [review]

File: cmake/external/abseil-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,10 @@ include(FetchContent)
 set(ABSL_PROPAGATE_CXX_STD 1)
 set(BUILD_TESTING 0)
 
+if(MSVC)
+add_compile_options(/WX-)
```
Comment (snnn) [inline cmake/external/abseil-cpp/CMakeLists.txt:11]: But who set "/WX"?  By default warnings are not errors. I only set "/WX" after all third-party libraries were built. i.e. after this file is included.

Comment (snnn) [reply cmake/external/abseil-cpp/CMakeLists.txt:11]: See https://github.com/microsoft/onnxruntime/blob/master/cmake/CMakeLists.txt#L1163 

"WX" is set at line 1163. But abseil is at line 111. Much before it.

**Review by snnn — CHANGES_REQUESTED** [review]
Sorry I don't think this change is needed.

File: cmake/external/abseil-cpp/CMakeLists.txt (hunk 1 above)
Comment (linnealovespie) [reply cmake/external/abseil-cpp/CMakeLists.txt:11]: "/WX" is set by the internal windowsAI build

**Comment by tiagoshibata** [comment]
> Sorry I don't think this change is needed.

We have /WX enabled in the internal WindowsAI build before we include ORT. We could do something like disable /WX, set onnxruntime_DEV_MODE=ON, include ORT and re-enable it, but it seemed easier to have ORT support builds with /WX.

**Comment by snnn** [comment]
But abseil is just one of the many 3rd-party dependencies. What if another one had the same issue? I can't guarantee all 3rd-party dependencies are all warning-free. Whenever we update our compiler(which happens monthly) or a dependency, it could break.

**Comment by tiagoshibata** [comment]
Fine, looking at the code I'm not sure why we don't use onnxruntime_DEV_MODE. I'll discuss with Linnea about refactoring our side.

File: cmake/external/abseil-cpp/CMakeLists.txt (hunk 1 above)
Comment (ryanlai2) [reply cmake/external/abseil-cpp/CMakeLists.txt:11]: Would it be better to use "target_compile_options" and specify a target?

Comment (snnn) [reply cmake/external/abseil-cpp/CMakeLists.txt:11]: @ryanlai2 absl has tens of targets.

**Pull request closed without merging**